### PR TITLE
Plane: holdoff auto takeoff till rudder released on rudder arming

### PR DESCRIPTION
**Pull request by tridge**
this fixes an issue with rudder arming on quadplanes where the vehicle may yaw rapidly during the takeoff if the user is using rudder arming
tested in RealFlight

### CODE REVIEW

File: ArduPlane/radio.cpp [hunk 1]
```diff
@@ -118,9 +118,14 @@ void Plane::init_rc_out_aux()
 */
 void Plane::rudder_arm_disarm_check()
 {
+    const int16_t rudder_in = channel_rudder->get_control_in();
+    if (rudder_in == 0) {
```
Comment (magicrub) [inline ArduPlane/radio.cpp:122]: Is ==0 too strict?

Comment (tridge) [reply ArduPlane/radio.cpp:122]: no, == 0 is exactly right. It is an integer, and == 0 means it is within the dead-zone. Any non-zero value means the rudder is not centered

**Comment by tridge** [comment]
> I wonder if this check could be incorporated into the throttle wait functionality to work across more VTOL modes?

I did consider that, but none of the other modes need it, as they use pilot throttle, and motors are idling until throttle is raised.
it really is just AUTO takeoff.
Note the PR also affects GUIDED auto-takeoff, and that is harmless, although it would be possible for a user to rudder arm then very quickly send a takeoff command and get yaw, so the fix is right for GUIDED, but in practice the separation of takeoff and arming for GUIDED means the issue won't happen

Another reason for not applying this to other modes is the user may have a bad rudder trim. We can't check rudder trim on arming with rudder. With AUTO takeoff this isn't too bad, but I really don't want modes like QLOITER to not be able to yaw right if the users rudder stick trim is off

File: ArduPlane/quadplane.cpp [hunk 2]
```diff
@@ -2902,12 +2902,29 @@ void QuadPlane::takeoff_controller(void)
...
     }
 
+    // don't takeoff up until rudder is re-centered after rudder arming
+    if (plane.arming.last_arm_method() == AP_Arming::Method::RUDDER &&
+        (takeoff_last_run_ms == 0 ||
+         now - takeoff_last_run_ms < 1000) &&
+        !plane.seen_neutral_rudder &&
+        spool_state <= AP_Motors::DesiredSpoolState::GROUND_IDLE) {
+        // start motor spinning if not spinning already so user sees it is armed
+        set_desired_spool_state(AP_Motors::DesiredSpoolState::GROUND_IDLE);
```
Comment (tridge) [inline ArduPlane/quadplane.cpp:2922]: if over 3 seconds then print a msg every 3s to release rudder?

Comment (tridge) [reply ArduPlane/quadplane.cpp:2931]: done

**Comment by rmackay9** [comment]
Tridge decided this was too complex a change to backport to 4.3

**Pull request merged**